pkg/image/importer/importer.go

```diff
@@ -509,6 +510,14 @@ func (isi *ImageStreamImporter) importRepositoryFromDocker(ctx gocontext.Context
 				continue
 			}
 		}
+
+		signatures, err := signature.GetSignatureForImage(repository.Registry.Hostname(), repository.Name, importDigest.Image.Name)
+		if err == nil {
+			importDigest.Image.Signatures = signatures
```

Comment:
> we don't update the images after we create them or track any changes to an image itself once imported.

The thing is, the signatures are intentionally not “a part of the image”, and may change independently.  An image originally published internally by a vendor as $registry/qa/product:2.1-rc5 may be republished (without any modification) and signed by the same vendor as $registry/released/product:2; then a third-party-integrator may republish it as a part of a larger software suite as $integrator/suite/infrastructure:5.8 .

Just because the image was previously visible with one set of signatures should not make it impossible to import a different set; and conversely just because the image has been recently re-imported with a fresh set of signatures should not remove the previously imported signatures.

I guess this is not _immediately_ blocking, but at the latest for the multi-tenant use, with different tenants importing images from different registries, this will need to be solved.

(And, eventually, we might want to think about marking signatures as obsolete when the authoritative registry stops serving them, so that a vendor can make a known-vulnerable image difficult to use. _Not_ removing such signatures, we need them to prove the image came from the vendor during forensic investigations. And we would have to define “authoritative registry” somehow.)